samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```
@@ -0,0 +1,498 @@
+      AdminUtils.deleteTopic(zkUtils(), kafkaTopic);
+    }
+    super.tearDown();
+  }
+
+  // Helper methods that are shared by tests.
+  private List<String> readProcessorIdsFromZK(String zkPathPrefix, List<String> processorIds) {
+    ZkClient zkClient = new ZkClient(zkConnect());
+    ZkKeyBuilder zkKeyBuilder = new ZkKeyBuilder(zkPathPrefix);
+    ZkUtils zkUtils = new ZkUtils(zkKeyBuilder, zkClient, 10000);
+    return zkUtils.getActiveProcessorsIDs(processorIds);
+  }
+
+  private String readJobModelVersionFromZK(String basePath) {
+    ZkUtils zkUtils = null;
+    try {
+      zkUtils = getZKUtils(basePath);
+      return zkUtils.getJobModelVersion();
+    } finally {
+      if (zkUtils != null) {
+        zkUtils.close();
+      }
+    }
+  }
+
+  private JobModel readJobModelFromZK(String basePath) {
+    ZkUtils zkUtils = null;
+    try {
+      zkUtils = getZKUtils(basePath);
+      zkUtils.connect();
+      return zkUtils.getJobModel(zkUtils.getJobModelVersion());
+    } finally {
+      if (zkUtils != null) {
+        zkUtils.close();
+      }
+    }
+  }
+
+  private ZkUtils getZKUtils(String basePath) {
+    ZkClient zkClient = new ZkClient(zkConnect());
+    ZkKeyBuilder zkKeyBuilder = new ZkKeyBuilder(basePath);
+    return new ZkUtils(zkKeyBuilder, zkClient, 10000);
+  }
+
+  private void publishKafkaEvents(String topic, int numMessages, String streamProcessorId) {
+    KafkaProducer producer = getKafkaProducer();
+    for (int messageIndex = 0; messageIndex < numMessages; messageIndex++) {
+      try {
+        LOGGER.info("Publish kafka event with index : {} for stream processor: {}.", messageIndex, streamProcessorId);
+        producer.send(new ProducerRecord(topic, new TestKafkaEvent(streamProcessorId, String.valueOf(messageIndex)).toString().getBytes())).get();
+      } catch (Exception  e) {
+        LOGGER.info("Exception when producing message to kafka topic: {}", new Object[]{topic, e});
```

Log format seems wrong, might swallow exception.